filters/include/pcl/filters/functor_filter.h

```diff
@@ -25,8 +26,8 @@ constexpr static bool is_functor_for_filter_v =
  * \ingroup filters
  */
 template <typename PointT, typename Functor>
-class FunctorFilter : public FilterIndices<PointT> {
-  using Base = FilterIndices<PointT>;
+class FunctorFilter : public FilterIndicesExecutor<PointT, FunctorFilter<PointT, Functor>> {
+  using Base = FilterIndicesExecutor<PointT, FunctorFilter<PointT, Functor>>;
```

Comment:
We've discussed the changes in inheritance before and demonstrated the need behind it a couple of times. Did you by any chance agglomerate all that info somewhere I can check? I'm asking just in case I'm lucky and you did :)

Reply:
After the last discussion, I have incorporated everything in #3